yb-voyager/cmd/importDataFileBatchProducer_test.go

```diff
@@ -0,0 +1,333 @@
+		defer os.RemoveAll(fmt.Sprintf("%s/", lexportDir))
+	}
+
+	fileContents := `id,val
+1, "hello"`
+	_, task, err := setupFileForTest(lexportDir, fileContents, ldataDir, "test_table")
+	assert.NoError(t, err)
+
+	batchproducer, err := NewFileBatchProducer(task, state)
+	assert.NoError(t, err)
+
+	assert.False(t, batchproducer.Done())
+
+	batch, err := batchproducer.NextBatch()
+	assert.NoError(t, err)
+	assert.NotNil(t, batch)
+	assert.Equal(t, int64(1), batch.RecordCount)
+	assert.True(t, batchproducer.Done())
+}
+
+func TestFileBatchProducerBasedOnRowsThreshold(t *testing.T) {
+	// max batch size in rows is 2
+	ldataDir, lexportDir, state, err := setupDependenciesForTest(2, 1024)
+	assert.NoError(t, err)
+
+	if ldataDir != "" {
+		defer os.RemoveAll(fmt.Sprintf("%s/", ldataDir))
+	}
+	if lexportDir != "" {
+		defer os.RemoveAll(fmt.Sprintf("%s/", lexportDir))
+	}
+
+	fileContents := `id,val
+1, "hello"
+2, "world"
+3, "foo"
+4, "bar"`
+	_, task, err := setupFileForTest(lexportDir, fileContents, ldataDir, "test_table")
+	assert.NoError(t, err)
+
+	batchproducer, err := NewFileBatchProducer(task, state)
+	assert.NoError(t, err)
+
+	assert.False(t, batchproducer.Done())
+
+	var batches []*Batch
+	for !batchproducer.Done() {
+		batch, err := batchproducer.NextBatch()
+		assert.NoError(t, err)
+		assert.NotNil(t, batch)
+		batches = append(batches, batch)
+	}
+
+	// 2 batches should be produced
+	assert.Equal(t, 2, len(batches))
+	// each of length 2
+	assert.Equal(t, int64(2), batches[0].RecordCount)
+	batchContents, err := os.ReadFile(batches[0].GetFilePath())
+	assert.NoError(t, err)
+	assert.Equal(t, "id,val\n1, \"hello\"\n2, \"world\"", string(batchContents))
+
+	assert.Equal(t, int64(2), batches[1].RecordCount)
+	batchContents, err = os.ReadFile(batches[1].GetFilePath())
+	assert.NoError(t, err)
+	assert.Equal(t, "id,val\n3, \"foo\"\n4, \"bar\"", string(batchContents))
+}
+
+func TestFileBatchProducerBasedOnSizeThreshold(t *testing.T) {
+	// max batch size in size is 25 bytes
+	maxBatchSizeBytes := int64(25)
+	ldataDir, lexportDir, state, err := setupDependenciesForTest(1000, maxBatchSizeBytes)
+	assert.NoError(t, err)
+
+	if ldataDir != "" {
+		defer os.RemoveAll(fmt.Sprintf("%s/", ldataDir))
+	}
+	if lexportDir != "" {
+		defer os.RemoveAll(fmt.Sprintf("%s/", lexportDir))
+	}
+
+	// each row exccept header is 10 bytes
+	fileContents := `id,val
+1, "abcde"
+2, "ghijk"
+3, "mnopq"
+4, "stuvw"`
+	_, task, err := setupFileForTest(lexportDir, fileContents, ldataDir, "test_table")
+	assert.NoError(t, err)
+
+	batchproducer, err := NewFileBatchProducer(task, state)
+	assert.NoError(t, err)
+
+	assert.False(t, batchproducer.Done())
+
+	var batches []*Batch
+	for !batchproducer.Done() {
+		batch, err := batchproducer.NextBatch()
+		assert.NoError(t, err)
+		assert.NotNil(t, batch)
+		batches = append(batches, batch)
+	}
+
+	// 3 batches should be produced
+	// while calculating for the first batch, the header is also considered
```

Comment:
Oh right, while preparing the first batch - we add the bytes of the header to the batch's total bytes but for the further batches, we don't as we already have the header we don't include it in the batch's bytes. 
I think worth testing if in some cases where the number of columns is huge can this header's bytes can also contribute to the batches' bytes and should be included. 
Can you please add a TODO while we are adding a header to bthe atch file to fix this if required?

Reply:
Yeah, did not want to change the implementation as part of this PR. Will add a TODO:

Reply:
@makalaaneesh @priyanshi-yb   i think we should be uniform across all the batches, either consider it in all or not consider at all, which we can discuss.